R/object_name_linters.R

```diff
@@ -44,15 +44,36 @@ object_name_xpath <- local({
 #' @param styles A subset of
 #'   \Sexpr[stage=render, results=rd]{lintr:::regexes_rd}. A name should
 #'   match at least one of these styles.
+#' @param regexes A (possibly named) character vector specifying a custom naming convention.
+#'   If named, the names will be used in the lint message. Otherwise, "custom" will be used as a name for the style.
+#'   Quotes (`` `"' ``) and specials (`%` and trailing `<-`) are not considered part of the name to be matched.
 #' @evalRd rd_tags("object_name_linter")
 #' @seealso [linters] for a complete list of linters available in lintr.
 #' @export
-object_name_linter <- function(styles = c("snake_case", "symbols")) {
-  styles <- match.arg(styles, names(style_regexes), several.ok = TRUE)
+object_name_linter <- function(styles = c("snake_case", "symbols"), regexes = character()) {
+  if (length(styles) > 0L) {
+    # Allow `object_name_linter(NULL, "my_regex")`
```

Comment:
not a big fan of `(NULL, "my_regex")` as a signature... can't we allow `object_name_linter(regex = "my_regex")` and use `missing()` on the arguments instead?

Reply:
I thought about it but it seemed confusing to me because we still want `object_name_linter()` to use the default `styles`, so behavior of `missing(styles)` depends on `missing(regexes)`.

Reply:
>  behavior of missing(styles) depends on missing(regexes).

I don't see a big problem with that...

Both look pretty clear to me in usage:

```r
object_name_linter() # use default 'styles'
object_name_linter(regex = "regex") # use regex, not 'styles'
```

Reply:
Okay.